### DISCUSSION TITLE
Need to ignore files on Sync Website to Git Repository

### DISCUSSION THREAD

**Discussion by nordseo** [GetPublii/Publii#1890] [Help]
Publii is replacing important files when deploying (Syncing) an update to Git Repository. 

1. We are not using Github pages.
2. We are using fleek.xyz which pulls our Github repository to on-chain file hosting, and IPFS.
3. We need to force Publii to ignore a file > `_redirects` (this is an IPFS formatted redirect file).
4. We also need Publii to ignore the `.gitignore` file. (otherwise, `.DS_Store` files keep getting deployed again to the repo automatically).

We have have tested other methods a little, and we are aware that Publii plans to remove direct Git support, but we think it should be kept if possible so our updates are deployed easier, and we don't ever plan to use centralized hosting such as Github Pages, or any kind of centralized hosting in the future. 

Currently, we have to perform a fetch, manually replacing the files that Publii deletes when the Sync is executed, and push a second commit for each update. Having a specific branch for Publii is also another option, but, is about the same amount of work each update anyways.

**Comment by dziudek** [comment]
@nordseo 

Direct Git support will be available in the future as is - we had plans to abandon Github Pages and Gitlab Pages support, but in the last months we have decided that we will still support these methods.

Regarding `_redirects` issue - you can put it in the File Manager under root directory. 

Regarding `.gitignore` - we cannot ignore this file, because it should be stored in the repository.

I think that for your requirements you can consider custom build - I did not use this platform, but I see that they have such option: https://fleek.xyz/docs/platform/deployments/

Then you can remove unnecessary files or replace content of `_redirects` file with a proper one

**Reply by nordseo** [reply]
Thanks for the tips. (late response).

I have decided to bypass GitHub to push my sites directly to Fleek.xzy with Fleek CLI. Also, I will attempt to make a plugin that will handle a proper merge into the Output Directory location using the Manual Deployment option.

I am currently using Radicle Git (radicle.xyz) for my repos now, so I still need Publii to avoid deleting non-Publii-related existing files in the repo, and only replace files that it generates on Sync export.

I'll follow up again here if my tests are successful. But, later on, maybe it would be a good idea to build the feature into Publii's core to avoid deleting other files from the destination folder?

I'm aware that I could export the files to another folder, then manually move them to replace the files, but this is yet another step I would like to eliminate, and also, the issue of previous Publii files would remain. So, the script I'm writing should do a proper merge, and only delete files that Publii originally created (not the `.git`, `fleek.json`, `_redirects`, and other files I keep in the root directory).
